sky/core.py

```diff
@@ -796,17 +796,21 @@ def spot_queue(refresh: bool,
             does not exist.
         RuntimeError: if failed to get the spot jobs with ssh.
     """
-
     stop_msg = ''
     if not refresh:
-        stop_msg = 'To view the latest job table: sky spot queue --refresh'
-    controller_status, handle = backend_utils.is_controller_up(
-        controller_type=controller_utils.Controllers.SPOT_CONTROLLER,
-        stopped_message=stop_msg)
+        stop_msg = ('No in-progress spot jobs.\nTo view the latest job table: '
+                    'sky spot queue --refresh')
+    try:
+        handle = backend_utils.is_controller_accessible(
+            controller_type=controller_utils.Controllers.SPOT_CONTROLLER,
+            stopped_message=stop_msg)
+    except exceptions.ClusterNotUpError as e:
+        if not refresh:
+            raise
+        handle = None
+        controller_status = e.cluster_status
 
-    if (refresh and controller_status in [
-            status_lib.ClusterStatus.STOPPED, status_lib.ClusterStatus.INIT
-    ]):
+    if refresh and handle is None:
```

Comment:
Is this right: When `refresh and handle is not None`, we have performed a refresh in is_controller_accessible() and don't refresh again here.

If so, worth a comment.

Reply:
No, when `handle is not None` we directly try to `ssh` into the controller to get the spot queue, but if `refresh and handle is None` here, we will try to restart the controller.